### PR TITLE
Solve most NEXT_MAJOR comments

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this is for NEXT MAJOR.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -29,10 +29,9 @@
         "sonata-project/block-bundle": "^3.20",
         "sonata-project/doctrine-extensions": "^1.8",
         "sonata-project/doctrine-orm-admin-bundle": "^3.19",
-        "sonata-project/form-extensions": "^0.1.1 || ^1.4",
-        "sonata-project/notification-bundle": "^3.8",
```
Comment (eerison) [inline composer.json]: you can not remove this yet!

Comment (jordisala1991) [reply composer.json]: Yep , I saw it. This PR will remain open, but as a draft.

Comment (eerison) [inline composer.json:33]: you can't remove this yet!

**Comment by eerison** [comment]
after this PR: #1434 pass, Can you search for `--site=all` into the doc and remove, this option will be remove on the next major

**Comment by jordisala1991** [comment]
Yes, tomorrow or on weekend I will manual merge 3.x to 4.x and try again to remove notification bundle.

**Comment by eerison** [comment]
@jordisala1991 I tried to use your branch But I got too many conflicts, then I decided to open a PR to remove the old code related with commands and notification-bundle, I guess it'll help you somehow :) 

#1460

**Comment by jordisala1991** [comment]
Probably this PR will never get merged, I already splitted 2 or 3 PRs from it.

File: src/Model/SnapshotChildrenCollection.php [hunk 2]
```diff
@@ -54,6 +54,10 @@ public function offsetSet($offset, $value): void
         $this->collection->offsetSet($offset, $value);
     }
 
+    /**
+     * @return mixed
+     */
+    #[\ReturnTypeWillChange]
```
Comment (eerison) [inline src/Model/SnapshotChildrenCollection.php:60]: Isn't it an php 8 feature?

Comment (jordisala1991) [reply src/Model/SnapshotChildrenCollection.php:60]: Yep, it count as a comment on PHP 7.

**Review by VincentLanglet — APPROVED** [review]

**Pull request merged**